packages/mds-transaction-service/repository/index.ts

```diff
@@ -148,7 +148,9 @@ export const TransactionRepository = ReadWriteRepository.Create(
           const queryBuilder = connection
             .getRepository(TransactionEntity)
             .createQueryBuilder(alias)
-            .select('provider_id, SUM(amount) as amount, COUNT(*) as count')
+            .select(
+              'provider_id, SUM(amount) as amount, COUNT(NULLIF(amount, 0)) as non_zero_amount_count, ABS(COUNT(*) - COUNT(NULLIF(amount, 0))) as zero_amount_count'
```

Comment:
This may also work and perhaps be a little easier to parse?
```sql
SUM(amount), 
SUM(CASE WHEN amount != 0 THEN 1 ELSE 0 END) AS non_zero_amount_count, 
SUM(CASE WHEN amount = 0 THEN 1 ELSE 0 END) AS zero_amount_count
```

Reply:
Nice idea, fixed in https://github.com/lacuna-tech/mds-core/pull/1049/commits/332f7274d67a15a0d3297a2b725004ddd7a6f2f1